broker/src/main/java/io/camunda/zeebe/broker/partitioning/PartitionFactory.java

```diff
@@ -170,7 +170,8 @@ List<ZeebePartition> constructPartitions(
               new PartitionProcessingState(owningPartition),
               diskSpaceUsageMonitor,
               gatewayBrokerTransport,
-              topologyManager);
```

Comment:
:thought_balloon: Just to clarify why this refactoring is made: we want to make the GatewayStreamer a new top-level service, i.e. the job stream service. We also need to expose the GatewayStreamer for the work you'll do on the broker/gateway side. So for these reasons, we need to move the instantiation up and pass it to the StreamPlatform, correct? 

>It's cumbersome to pass the job streamer from the broker down to the job state. I think this highlights that it's weird that the job state in general should take care of it. I think your team will have a better idea of how to properly refactor this.

I see what you mean when you say that. :smile: I would say that your refactoring makes the code cleaner, but then we get [this](https://github.com/camunda/zeebe/pull/11889/files#diff-71105e453b58e4b990bde2e746beaa2fe0cef0ae1faa45024deea37e8b06374bR78) which feels a bit weird.

I think it's good for now. I will look into what we can improve when I start the task breakdown for the work on the engine side within this epic.